### DISCUSSION TITLE
missing FROM-clause entry for table "chapter"

### DISCUSSION THREAD

**Discussion by Electric-Dragon** [supabase/supabase#6456] [Questions]
I am trying to return the sum of certain values that are returned from a function call on a specific row

This is the main function which is called on the client (`get_series_total_likes`):
```
DECLARE
  num integer;
BEGIN
  SELECT sum(get_chapter_likes(chapter)) 
    into num 
  FROM public.chapter_likes
  WHERE chapter.seriesid = series_id;

  RETURN num;

END; 
```

This is `get_chapter_likes`:
```
DECLARE
  num integer;
BEGIN
  SELECT count(*) 
    into num 
  FROM public.chapter_likes
  WHERE chapter = chapterid;

  RETURN num;

END; 
```

The foreign key on `chapter_likes.chapter` references `chapters.id` and `chapters.seriesid` references `series.id`

I tried using a join and it kind of works, but I have no idea on how to integrate it with the main function:
```
SELECT chapter_likes.id, chapters.id
From chapter_likes
INNER JOIN chapters ON chapter_likes.chapter = chapters.id
WHERE chapters.seriesid = series_id
```

**Comment by Electric-Dragon** [comment]
@steve-chavez can you help me out?

**Reply by steve-chavez** [reply]
> I tried using a join and it kind of works, but I have no idea on how to integrate it with the main function

For now, I suggest to look at https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-inner-join.

**Comment by steve-chavez** [comment]
@Electric-Dragon What's the definition of `chapter_likes`?

**Reply by Electric-Dragon** [reply]
@steve-chavez 

Here you go, the `chapter_likes` table

```
CREATE TABLE IF NOT EXISTS public.chapter_likes
(
    id bigint NOT NULL GENERATED BY DEFAULT AS IDENTITY ( INCREMENT 1 START 1 MINVALUE 1 MAXVALUE 9223372036854775807 CACHE 1 ),
    "user" uuid NOT NULL,
    chapter uuid NOT NULL,
    CONSTRAINT "chapterLikes_pkey" PRIMARY KEY (id),
    CONSTRAINT "chapterLikes_user_fkey" FOREIGN KEY ("user")
        REFERENCES auth.users (id) MATCH SIMPLE
        ON UPDATE NO ACTION
        ON DELETE NO ACTION,
    CONSTRAINT chapter_likes_chapter_fkey FOREIGN KEY (chapter)
        REFERENCES public.chapters (id) MATCH SIMPLE
        ON UPDATE NO ACTION
        ON DELETE NO ACTION
)
```

---

And this is the `chapters` table

```
CREATE TABLE IF NOT EXISTS public.chapters
(
    id uuid NOT NULL DEFAULT uuid_generate_v4(),
    title character varying(25) COLLATE pg_catalog."default" DEFAULT 'Untitled'::character varying,
    body jsonb,
    images text[] COLLATE pg_catalog."default",
    seriesid uuid NOT NULL,
    chapternum integer NOT NULL,
    is_published boolean NOT NULL DEFAULT false,
    createdat timestamp without time zone NOT NULL DEFAULT now(),
    totalreads integer NOT NULL DEFAULT 0,
    CONSTRAINT chapters_pkey PRIMARY KEY (id),
    CONSTRAINT chapters_seriesid_fkey FOREIGN KEY (seriesid)
        REFERENCES public.series (id) MATCH SIMPLE
        ON UPDATE NO ACTION
        ON DELETE NO ACTION
)
```

**Comment by steve-chavez** [comment]
> DECLARE
  num integer;
BEGIN
  SELECT sum(get_chapter_likes(chapter)) 
    into num 
  FROM public.chapter_likes
  WHERE chapter.seriesid = series_id;
  RETURN num;
END; 

The error comes from `WHERE chapter.seriesid = series_id;` - there's no 'chapter' table.

**Reply by Electric-Dragon** [reply]
I fixed this but it still says `missing FROM-clause entry for table "chapters"`

**Reply by Electric-Dragon** [reply]
@steve-chavez any updates?

**Reply by steve-chavez** [reply]
@Electric-Dragon Since this is a pure SQL issue(not directly related to a Supabase component), let me ping @GaryAustin1 and @silentworks for further help.

**Reply by GaryAustin1** [reply]
At a minimum it would help to see where code/table is at currently.   I would also suggest sometimes the discord help server is a better place for more general help as there seem to be more users wandering thru there.  In particular there are a few SQL gurus over there  that help people out frequently.   https://discord.com/channels/839993398554656828/869405720934744086

**Reply by Electric-Dragon** [reply]
Okay...I'll give the server a try
